test/CMakeLists.txt

```diff
@@ -11,26 +11,18 @@ set(LLVM_SPIRV_TEST_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 pkg_search_module(SPIRV_TOOLS SPIRV-Tools)
 if(SPIRV_TOOLS_FOUND)
   find_program(SPIRV_TOOLS_SPIRV_AS NAMES spirv-as PATHS ${SPIRV_TOOLS_PREFIX}/bin)
-  if(SPIRV_TOOLS_SPIRV_AS)
-    set(SPIRV_TOOLS_SPIRV_AS_FOUND True)
-  endif()
 
   find_program(SPIRV_TOOLS_SPIRV_VAL NAMES spirv-val PATHS ${SPIRV_TOOLS_PREFIX}/bin)
-  if(SPIRV_TOOLS_SPIRV_VAL)
-    set(SPIRV_TOOLS_SPIRV_VAL_FOUND True)
-  endif()
 
   set(SPIRV_TOOLS_BINDIR "${SPIRV_TOOLS_PREFIX}/bin")
 endif()
 
 if(NOT SPIRV_TOOLS_SPIRV_AS)
-  message(WARNING "spirv-as not found! SPIR-V assembly tests will not be run.")
-  set(SPIRV_TOOLS_SPIRV_AS_FOUND False)
+  message(FATAL_ERROR "spirv-as not found!")
```

Comment:
I think that if we a bit afraid that downstream users are not ready to this changes then we could design them in a way that they could be easily reverted by the users: I mean, let's by default have strong dependency on SPIRV-Tools, but also let's leave an option which allows to make this dependency optional and partially disable some tests.

In this case we can commit this patch sooner and remove this option later (let's say in a month or so from PR merge)

```
$ cmake -DLLVM_ENABLE_TESTING=ON ..
...
FATAL_ERROR: spirv-tools are required to test the translator.
For transition period you can specify 'SPIRV_TOOLS_FORCE_DISABLE=ON' to skip these tests.
Note: this option is only intended for transition period and will be removed in the upcoming revisions of the translator.
cmake failed
$ cmake -DLLVM_ENABLE_TESTING=ON -DSPIRV_TOOLS_FORCE_DISABLE=ON ..
...
WARNING: You are using 'SPIRV_TOOLS_FORCE_DISABLE' which will be removed in the upcoming revisions of the translator.
Please configure your environment to include SPIRV-Tools as well to test the translator.
Some tests will be disabled for now.
cmake passed
```

I understand that such change is basically almost a "revert" of this patch, but we can look at this as an another way of communicating such important upcoming change for our downstream users.

@svenvh, @StuartDBrady, @AlexeySotkin, @bader, what do you think?

Reply:
Sounds good to me.

Reply:
Sounds reasonable, I'll prepare an alternative patch.

Reply:
@svenvh, One more idea from @AlexeySotkin: right now this patch relies on `LLVM_INCLUDE_TESTS` and basically, there is no way for user to disable tests only for the translator.

If we add something like `LLVM_SPIRV_INCLUDE_TESTS` to add possibility for users to disable testing specifically for the translator, then we probably can go with this patch as-is (almost).

Summarizing, two options:
* rely on `LLVM_INCLUDE_TESTS`, but add possibility to temporary disable new hard requirement for SPIRV-Tools
* rely on separate `LLVM_SPRIV_INCLUDE_TESTS` - this patch can be left almost as-is then

Reply:
Looks like it is a common practice in the LLVM project to have an option to enable/disable tests for subprojects. For example, there are [CLANG_INCLUDE_TESTS](https://github.com/llvm/llvm-project/blob/559ae14c9b6cf203e0e7495892080e52e89c1ea0/clang-tools-extra/CMakeLists.txt#L17), 
[LIBCXX_INCLUDE_TESTS](https://github.com/llvm/llvm-project/blob/80ef2f05d878fd6b00bbfb2ed1c04e2657d241cc/libcxx/CMakeLists.txt#L82) and [LLDB_INCLUDE_TESTS](https://github.com/llvm/llvm-project/blob/ff5982aa91d1ca0c11e00b3d2b22d32c66bb8590/lldb/CMakeLists.txt#L72)

Reply:
> rely on separate `LLVM_SPRIV_INCLUDE_TESTS` - this patch can be left almost as-is then

Introducing `LLVM_SPIRV_INCLUDE_TESTS` and this patch does not allow partially disabling translator tests.  But it will allow disabling the entire translator test suite while still allowing to run Clang/LLVM tests.  Do you still want to have a transition period before making SPIRV-Tools a hard dependency (to allow partially running the test suite in the mean time)?

I agree that `LLVM_SPIRV_INCLUDE_TESTS` is a good thing to have, so I've opened https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/335

Reply:
@svenvh, personally, I think that if we have a separate option to enable/disable testing of the translator, then we don't need to invent any workarounds or transition periods.

So, from my point of view, I would go with #335 first and then this PR can be committed as-is: any user who wants to test the translator must get SPIRV-Tools in his environment and at the same time he can disable the translator testing at all for the transition period if such is required.

Comparing to that, `SPIRV_TOOLS_FORCE_DISABLE`, looks like a hack :)

Reply:
> @svenvh, personally, I think that if we have a separate option to enable/disable testing of the translator, then we don't need to invent any workarounds or transition periods.
> 

Fair enough, I have updated the fatal error message to mention how llvm-spirv tests can be disabled.

Keeping open for a bit longer such that any other objections can be raised before merging.